include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+ * You must not reference the texture after calling this function.
+ *
+ * \param device a GPU context
+ * \param sampler a sampler to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseSampler(
+    SDL_GpuDevice *device,
+    SDL_GpuSampler *sampler);
+
+/**
+ * Frees the given buffer as soon as it is safe to do so.
+ * You must not reference the buffer after calling this function.
+ *
+ * \param device a GPU context
+ * \param buffer a buffer to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseBuffer(
+    SDL_GpuDevice *device,
+    SDL_GpuBuffer *buffer);
+
+/**
+ * Frees the given transfer buffer as soon as it is safe to do so.
+ * You must not reference the transfer buffer after calling this function.
+ *
+ * \param device a GPU context
+ * \param transferBuffer a transfer buffer to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseTransferBuffer(
+    SDL_GpuDevice *device,
+    SDL_GpuTransferBuffer *transferBuffer);
+
+/**
+ * Frees the given compute pipeline as soon as it is safe to do so.
+ * You must not reference the compute pipeline after calling this function.
+ *
+ * \param device a GPU context
+ * \param computePipeline a compute pipeline to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseComputePipeline(
+    SDL_GpuDevice *device,
+    SDL_GpuComputePipeline *computePipeline);
+
+/**
+ * Frees the given shader as soon as it is safe to do so.
+ * You must not reference the shader after calling this function.
+ *
+ * \param device a GPU context
+ * \param shader a shader to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseShader(
+    SDL_GpuDevice *device,
+    SDL_GpuShader *shader);
+
+/**
+ * Frees the given graphics pipeline as soon as it is safe to do so.
+ * You must not reference the graphics pipeline after calling this function.
+ *
+ * \param device a GPU context
+ * \param graphicsPipeline a graphics pipeline to be destroyed
+ *
+ * \since This function is available since SDL 3.x.x
+ */
+extern SDL_DECLSPEC void SDLCALL SDL_GpuReleaseGraphicsPipeline(
+    SDL_GpuDevice *device,
+    SDL_GpuGraphicsPipeline *graphicsPipeline);
+
+/*
+ * COMMAND BUFFERS
+ *
+ * Render state is managed via command buffers.
+ * When setting render state, that state is always local to the command buffer.
+ *
+ * Commands only begin execution on the GPU once Submit is called.
+ * Once the command buffer is submitted, it is no longer valid to use it.
+ *
+ * In multi-threading scenarios, you should acquire and submit a command buffer on the same thread.
+ * As long as you satisfy this requirement, all functionality related to command buffers is thread-safe.
```

Comment:
Is this being tested in a stress-test somewhere?

Reply:
@thatcosmonaut might be using this but we don't have a test for this ourselves; I can check with some FNA titles but the replays are all synchronous so I'll need to finish C# support first.